contracts/ronin/StakingVesting.sol

```diff
@@ -104,6 +124,16 @@ contract StakingVesting is
 
     blockProducerBonus = forBlockProducer ? blockProducerBlockBonus(block.number) : 0;
     bridgeOperatorBonus = forBridgeOperator ? bridgeOperatorBlockBonus(block.number) : 0;
+
+    if (!_isREP10Activated) {
+      uint256 currPeriod = IRoninValidatorSet(getContract(ContractType.VALIDATOR)).currentPeriod();
+      if (currPeriod >= _rep10ActivationPeriod) {
+        _isREP10Activated = true;
+        _fastFinalityRewardPercentage = _fastFinalityRewardPercentageREP10;
+        emit REP10FastFinalityRewardActivated(currPeriod, _fastFinalityRewardPercentage);
```

Comment:
Prefer using _setFastFinality method for emitting existing event. Emit another one if needed for debug the transition.